newIDE/app/src/MainFrame/index.js

```diff
@@ -2089,6 +2092,24 @@ const MainFrame = (props: Props) => {
           if (!saveAsLocation) {
             return; // Save as was cancelled.
           }
+          const fileIdentifier = saveAsLocation.fileIdentifier;
```

Comment:
Just a code style thing: let's put this in another file to avoid adding any more code to MainFrame which already deals with too many things.